support/Python/Schedule.py

```diff
@@ -0,0 +1,704 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+import re
+import shutil
+import subprocess
+from pathlib import Path
+from typing import Optional, Sequence, Union
+
+import click
+import jinja2
+import jinja2.meta
+import yaml
+from spectre.support.DirectoryStructure import (Checkpoint, Segment,
+                                                list_checkpoints,
+                                                list_segments)
+
+logger = logging.getLogger(__name__)
+
+# CMake configures the submit script templates for the current machine to this
+# path
+default_submit_script_template = Path(__file__).parent / 'Submit.sh'
+
+
+def _resolve_executable(executable: Union[str, Path]) -> Path:
+    """Look up the 'executable' in the PATH
+
+    Raises 'ValueError' if the executable is not found.
+    """
+    logger.debug(f"Resolving executable: {executable}")
+    which_exec = shutil.which(executable)
+    if not which_exec:
+        raise ValueError(f"Executable not found in PATH: {executable}")
+    logger.debug(f"Found executable in PATH: {which_exec}")
+    return Path(which_exec).resolve()
+
+
+def _write_or_overwrite(text: str,
+                        path: Path,
+                        error_hint: Optional[str] = None,
+                        force: bool = False):
+    """Write the 'text' to a file at 'path'
+
+    Raise an 'OSError' if the file already exists, unless called with 'force'
+    or if the existing file content is identical to the 'text'.
+    The 'hint' is appended to the error message.
+    """
+    if path.exists():
+        if path.read_text() == text:
+            return
+        if not force:
+            raise OSError(f"File already exists at '{path}'. Delete it or "
+                          "retry with 'force' ('--force' / '-f')." +
+                          (("\n" + error_hint) if error_hint else ""))
```

Comment:
[optional] If we were confident that deleting the file was the correct thing to do, we would just do it here, so telling the user to do so may be giving bad instructions.  I'd just leave it at "File already exists.  Retry with --force to overwrite.".